lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -3803,11 +3834,12 @@ namespace Js
 
             scriptContext->GetThreadContext()->SetIsProfilingUserCode(isUserCode); // Update IsProfilingUserCode state
         }
+#endif // ENABLE_SCRIPT_PROFILING
 
         Var aReturn = NULL;
         JavascriptMethod origEntryPoint = function->GetFunctionInfo()->GetOriginalEntryPoint();
 
-        __try
+        __TRY_FINALLY_BEGIN // SEH is not guaranteed, see the implementation
```

Comment:
@jianchun I also didn't observe any SEH specific code. 

2 things I've been trying to to with this PR; 1) do not update any Windows specific code unless it is a bug etc. 2) generally update TRY/FINALLY code to macro.

The second one was out of the scope and we hadn't reach to a consensus there. Updating this implementation very close to original while adding a notice. In case, in the future, if someone expects this part of the code handling SEH, the comment above should be enough to keep the intentions away. In the meantime, adding a macro for future use.